### PR TITLE
Add null checking

### PR DESCRIPTION
**Pull request by stephen-riley**

### CODE REVIEW

File: Models/Sequence.cs [hunk 1]
```diff
@@ -34,6 +34,7 @@ public Sequence(BamAlignment bamAlignment)
             Identifier = new ReadOnlyMemory<byte>(bamAlignment.read_name).ToArray();
             Read = new ReadOnlyMemory<byte>(bamAlignment.seq).ToArray();
             Quality = new ReadOnlyMemory<byte>(bamAlignment.qual).ToArray();
+            Blank = Array.Empty<byte>();
```
Comment (michaeljon) [inline Models/Sequence.cs:37]: this is likely better anyway, the "blank" doesn't always mean blank, it's sometimes a duplicate of the identity line

File: Ovation.FasterQC.Net.csproj [hunk 2]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net6.0</TargetFramework>
+    <Nullable>enable</Nullable>
```
Comment (michaeljon) [inline Ovation.FasterQC.Net.csproj:6]: i'll bet this was a pain... that's why i don't do it

File: Readers/FastqLineReader.cs [hunk 3]
```diff
@@ -61,10 +61,10 @@ public bool ReadSequence(out Sequence sequence)
                     return false;
                 }
```
Comment (michaeljon) [inline Readers/FastqLineReader.cs:63]: hmm.... an empty read here really should blow up

**Review by michaeljon — APPROVED** [review]

**Comment by michaeljon** [comment]
FWIW I'm not sold on the value of nullable type checking for reference types. It's an additional layer that seems to get in the way of actually thinking about the solution.

File: Ovation.FasterQC.Net.csproj (hunk 2 above)
Comment (stephen-riley) [reply Ovation.FasterQC.Net.csproj:6]: yes, yes it was 😂

File: Readers/FastqLineReader.cs (hunk 3 above)
Comment (stephen-riley) [reply Readers/FastqLineReader.cs:63]: like throw an `EndOfStreamException`?

**Pull request merged**